### ISSUE TITLE
Build failure on macOS with Dune master

### ISSUE THREAD

**Issue by aantron** [ocaml/dune#1633]
We are [seeing](https://travis-ci.org/ocaml/odoc/jobs/466119777#L2152) this in odoc's CI, in the macOS row:

```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
[ERROR] The compilation of dune failed at "/Users/travis/.opam/4.07.0/.opam-switch/build/dune.1.6.2/./boot.exe --release -j 1".
#=== ERROR while compiling dune.1.6.2 =========================================#
# context              2.0.1 | macos/x86_64 | ocaml-base-compiler.4.07.0 | pinned(git+https://github.com/ocaml/dune.git#e583c0cc)
# path                 ~/.opam/4.07.0/.opam-switch/build/dune.1.6.2
# command              ~/.opam/4.07.0/.opam-switch/build/dune.1.6.2/./boot.exe --release -j 1
# exit-code            2
# env-file             ~/.opam/log/dune-11259-b76d46.env
# output-file          ~/.opam/log/dune-11259-b76d46.out
### output ###
# executable program file not found
```

The version number is misleading. Dune 1.6.2 release [installs](https://travis-ci.org/ocaml/odoc/jobs/466119777#L1153) fine. We are actually [using](https://travis-ci.org/ocaml/odoc/jobs/466119777#L2122) Dune `master` at that point in CI.

**Comment by rgrinberg** [comment]
I believe this error is caused by the fd's being exhausted on OSX. @jonludlam is looking into it.

**Comment by jonludlam** [comment]
For a quick workaround, `ulimit -n 512` worked for me. I'm looking into the root cause.

**Comment by diml** [comment]
Should be fixed by #1635

**Comment by aantron** [comment]
Thanks. odoc CI now passing.
